Update and-or.css

Christopher-O opened the pull request:
updated line/border from background-image to border-image in order for line to show up better in high contrast mode.

Garneauma commented:
Pre-approved upon successful review and assuming it meets all méli-mélo update requirements.

Garneauma commented:
Pre-approved upon successful review and completion of all work items.

Christopher-O commented:
Waiting for @duboisp  for conversion of WCAG accessibility points to HTML,
unsure exactly what this task entails
"Cross-link all the project pages and documents to ensure we can find all the information published via our GCWeb website."
@Garneauma  could you please advise?

duboisp commented on méli-mélo/2021-05-conjunction/and-or.css:
```diff
@@ -1,263 +1,286 @@
-[class*=cnjnctn-type-].cnjnctn-xs>[class*=cnjnctn-col]:not(:first-child) {
-    margin-left: 1.4em;
-    position: relative;
+@media (prefers-contrast: more) {
+  [class*=cnjnctn-type-].cnjnctn-xs > [class*=cnjnctn-col]:not(:first-child):after {
+    border-image: linear-gradient(to bottom, #ffffff 0.3em, #ffffff 0.3em, transparent 0.3em, transparent 2.35em, #ffffff 2.35em, #ffffff 2.35em) 1 100%;
+  }
```
I tested that CSS `prefers-contrast: more` and my testing showed it acceptable in our currently supported browser list.

Here a summary of my test result:

* Windows Chrome: work as expected
* Windows Edge: work as expected
* Windows Firefox dev: It is broken, the user can read the conjunction word "and" and "or" because there is a vertical black bar in the middle
* MacOS Big Sur Chrome: Don't work as expected, but the text is readable, the white border gradient only affect the circle around the word, not the word itself.
* MacOS Big Sur Safari: Don't work as expected, but the text is readable, the white border gradient only affect the circle around the word, not the word itself.

I am going to attache those screen capture as a comment in the PR, as this inline comment are only informative

duboisp commented:
Comment regarding the adaptive style for the **high contrast mode**, 

```css
@media (prefers-contrast: more) {
```

I tested that CSS and my testing showed it acceptable in our currently supported browser list.

The supported browser are:
* Chrome (Desktop and mobile)
* Safari (Desktop and mobile)
* Edge (Desktop only)

Here my test result in high contrast mode:

---

**Windows Chrome:** work as expected

![xs-chrome](https://github.com/wet-boew/GCWeb/assets/1676030/bbad8ba4-b416-49da-8e21-ee7b4133b38f)

---

**Windows Edge:** work as expected

Same visual as Windows Chrome

---

**Windows Firefox dev:** It is broken, the user can not clearly read the conjunction word "and" and "or" because there is a vertical black bar in the middle

![xs](https://github.com/wet-boew/GCWeb/assets/1676030/14dd52ea-7aa6-4606-9ae0-a6505c4b8b5d)

---

**MacOS Big Sur Chrome:** Don't work as expected, but the text is readable, the white border gradient only affect the circle around the word, not the word itself.

<img width="937" alt="Screen Shot 2024-06-18 at 9 35 49 PM" src="https://github.com/wet-boew/GCWeb/assets/1676030/9234b32b-b925-49dd-92bb-6fdda10b6de0">

---

**MacOS Big Sur Safari:** Don't work as expected, but the text is readable, the white border gradient only affect the circle around the word, not the word itself.

Same visual as MacOS Big Sur Chrome

---

**Here what it look like in normal mode**

![image](https://github.com/wet-boew/GCWeb/assets/1676030/2a14e733-3971-4130-a907-f79b53844dec)

---

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+    "earl:result": [
+        {
+            "@id": "_:sc_1.1.1",
+            "earl:test": "WCAG21:non-text-content(Level A)",
```
The id must match the success criteria ID on the W3C WCAG 2.1 technical specification page. The prefix `WCAG21:` is an alias to that technical specification.

Appending "(Level A)" make that a 404 not found link.

```suggestion
            "earl:test": "WCAG21:non-text-content",
```

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        },
+        {
+            "@id": "_:sc_1.2.1",
+            "earl:test": "WCAG21:audio-only-and-video-only(Level A)",
```
Must match the ID in the W3C specification of WCAG 2.1

```suggestion
            "earl:test": "WCAG21:audio-only-and-video-only-prerecorded",
```

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        },
+        {
+            "@id": "_:sc_1.2.2",
+            "earl:test": "WCAG21:captions-prerecorded(Level A)",
```
```suggestion
            "earl:test": "WCAG21:captions-prerecorded",
```

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        },
+        {
+            "@id": "_:sc_1.2.3",
+            "earl:test": "WCAG21:audio-description-or-media-alternative-prerecorded(Level A)",
```
```suggestion
            "earl:test": "WCAG21:audio-description-or-media-alternative-prerecorded",

```

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        },
+        {
+            "@id": "_:sc_1.2.4",
+            "earl:test": "WCAG21:captions-live(Level AA)",
```
```suggestion
            "earl:test": "WCAG21:captions-live",
```

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        },
+        {
+            "@id": "_:sc_1.2.5",
+            "earl:test": "WCAG21:audio-description-prerecorded(Level AA)",
```
```suggestion
            "earl:test": "WCAG21:audio-description-prerecorded",

```

duboisp commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        "dct:description": "Assess the GCWeb meli-melo conjunction-and_or page according to Standard on Web Accessibility (SWA)",
+        "earl:pointer": {
+            "@type": "oa:CssSelector",
+            "@value": "conjunction-and_or"
```
```suggestion
            "@value": ".cnjnctn-type-or,.cnjnctn-type-and"
```

duboisp commented:
@ouafaaetta as discussed, please submit a request change to the little update required in the accessibility report in JSON-LD. Thanks

duboisp commented:
Completed the review and local testing, everything look good. There is a few typo change to apply prior to merge.

Note that some additional change would be required, post merge like:
* Use jekyll front-matter to link the other page language
* Adding the 2 HTML placeholder page to render the accessibility assessment in an HTML format.
* Producing the index.json-ld file for documentation by leveraging the documentation provided.

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.2.2",
+            "earl:test": "WCAG21:on-input(Level A)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.2.2",
            "earl:test": "WCAG21:on-input",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.2.3",
+            "earl:test": "WCAG21:consistent-navigation(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.2.3",
            "earl:test": "WCAG21:consistent-navigation",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.2.4",
+            "earl:test": "WCAG21:consistent-identification(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.2.4",
            "earl:test": "WCAG21:consistent-identification",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.3.1",
+            "earl:test": "WCAG21:error-identification(Level A)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.3.1",
            "earl:test": "WCAG21:error-identification",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.3.2",
+            "earl:test": "WCAG21:labels-or-instructions(Level A)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.3.2",
            "earl:test": "WCAG21:labels-or-instructions",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.3.3",
+            "earl:test": "WCAG21:error-suggestion(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.3.3",
            "earl:test": "WCAG21:error-suggestion",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_3.3.4",
+            "earl:test": "WCAG21:error-prevention-legal-financial-data)(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_3.3.4",
            "earl:test": "WCAG21:error-prevention-legal-financial-data",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_4.1.1",
+            "earl:test": "WCAG21:parsing(Level A)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:Pass",
+            "dct:description": "Design uses standard HTML structure with a CSS overlay",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_4.1.1",
            "earl:test": "WCAG21:parsing",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:Pass",
            "dct:description": "Design uses standard HTML structure with a CSS overlay",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_4.1.2",
+            "earl:test": "WCAG21:Name, Role, Value (Level A)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "@id": "_:sc_4.1.2",
            "earl:test": "WCAG21:Name, Role, Value ",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            ]
+        },
+        {
+            "@id": "_:sc_4.1.3",
+            "earl:test": "WCAG21:status-messages(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
```
```suggestion
            "@id": "_:sc_4.1.3",
            "earl:test": "WCAG21:status-messages",
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
```
```suggestion
            "earl:mode": "earl:manual",
            "@type": [

```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
```
```suggestion
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        {
+            "@id": "_:sc_1.2.3",
+            "earl:test": "WCAG21:audio-description-or-media-alternative-prerecorded(Level A)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
+        },
```
```suggestion
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
        },
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+            "earl:test": "WCAG21:captions-live(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
```
```suggestion
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
```

ouafaaetta commented on méli-mélo/2021-05-conjunction/a11y-1-conjunction.json:
```diff
@@ -0,0 +1,651 @@
+        {
+            "@id": "_:sc_1.2.5",
+            "earl:test": "WCAG21:audio-description-prerecorded(Level AA)",
+            "earl:subject": "_:subject",
+            "earl:outcome": "earl:inapplicable",
+            "dct:description": "As defined",
+            "earl:mode": "earl:semiAuto",
+            "@type": [
+                "earl:TestResult",
+                "earl:Assertion"
+            ]
```
```suggestion
            "earl:subject": "_:subject",
            "earl:outcome": "earl:inapplicable",
            "dct:description": "As defined",
            "earl:mode": "earl:manual",
            "@type": [
                "earl:TestResult",
                "earl:Assertion"
            ]
```

duboisp reviewed (approved):
Reviewed and locally tested. Some typos has been also fixed and the implementation plan updated.
This experimental project is now ready for being stabilized in GCWeb. We will perform a full review soon.

Christopher-O commented:
Changes in the PR

- Changed border to appear in high contrast
- Changed font size of CSS generated text to em value for better scalability between parent font sizes
- Increased miminum height of side by side pattern to create consistant border above and below CSS text design
- Updated governance towards stabilization 

Impact for department

- No change from a departmental perspective

Impact for the public

- Allows border to be shown in high contrast (could require additional adjustment for FF and MacOS browsers)
- Fixed issue of using design in font sizes other than 20px (GCWeb standard)
